File: two-pointers/443/443.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        read = write = 0
        while read < len(chars):
            currChar, currCount = chars[read], 0
            while read < len(chars) and chars[read] == currChar:
                currCount += 1
                read += 1
            chars[write] = currChar
            write += 1
            if currCount > 1:
                for d in str(currCount):
                    chars[write] = d
                    write += 1
        return write

File: two-pointers/443/443.js
var compress = function(chars) {
  let read = (write = 0);
  while (read < chars.length) {
    let currChar = chars[read];
    let currCount = 0;
    while (read < chars.length && chars[read] === currChar) {
      currCount++;
      read++;
    }
    chars[write] = currChar;
    write++;
    if (currCount > 1) {
      for (let d of String(currCount).split("")) {
        chars[write] = d;
        write++;
      }
    }
  }
  return write;
};
